cms/server/admin/handlers/views.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+# Contest Management System - http://cms-dev.github.io/
+# Copyright © 2015 Stefano Maggiolo <s.maggiolo@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""UI fragments for AdminWebServer
+
+"""
+
+
+import tornado.web
+
+
+class ReevaluationButtons(tornado.web.UIModule):
+    def render(self,
+               url_root,
+               contest_id=None,
+               submission_id=None,
+               dataset_id=None,
+               participation_id=None,
+               user_id=None):
+        """Render reevaluation buttons for the given filters.
+
+        Apart from contest_id, the parameters are mutually exclusive
+        (i.e., only one can be provided). The only exception is when
+        providing a submission, in which case a specific dataset can
+        also be passed (the live one is used by default).
+
+        url_root (unicode): path to the root of the server.
+        contest_id (int): the id of the contest containing the
```

Comment:
Can contest_id be None? Here you say no, but you give it a None default value, and in activate_dataset.html in fact you don't pass any explicit contest_id